core-library/component-styling/indicator/_indicator.scss

```diff
@@ -0,0 +1,262 @@
+@use "../../typography/typography" as typography;
+@use "../../tokens/text" as text;
+@use "../../tokens/partial/focus" as focus;
+@use "sass:map";
+
+@mixin selector {
+  lib-indicator#{template-const.$escape} {
+    @content;
+  }
+}
+
+@mixin create {
+  @include selector() {
+    @include layout();
+  }
+}
+
+@mixin size-sm {
+  .text {
+    padding: 2px 8px;
+    border-radius: 11px;
+
+    @include typography.fontSize(map.get(text.$body, level), 4);
+
+    &.rounded {
+      min-width: 8px;
+    }
+
+    > svg {
+      width: 12px;
+    }
+  }
+
+  .number {
+    padding: 6px 9px;
+
+    > span {
+      min-width: 14px;
+    }
+
+    @include typography.fontSize(map.get(text.$body, level), 4);
+
+    &.num-lg {
+      padding: 2px 8px;
+      border-radius: 11px;
+    }
+  }
+
+  .dot {
+    width: 6px;
+    height: 6px;
+  }
+}
+
+@mixin size-lg {
+  .text {
+    padding: 4px 8px;
+    border-radius: 15px;
+
+    @include typography.fontSize(map.get(text.$body, level), 2);
+
+    &.rounded {
+      padding: 2px 8px;
+      min-width: 12px;
+    }
+
+    > svg {
+      width: 16px;
+    }
+  }
+
+  .number {
+    padding: 7px 10px;
+
+    > span {
+      min-width: 18px;
+    }
+
+    @include typography.fontSize(map.get(text.$body, level), 2);
+
+    &.num-lg {
+      padding: 4px 8px;
+      border-radius: 15px;
+    }
+  }
+
+  .dot {
+    width: 10px;
+    height: 10px;
+  }
+}
+
+@mixin status-color {
+  &.info,
+  &.teal {
+    color: var(--information-text);
+  }
+
+  &.warning,
+  &.orange {
+    color: var(--warning-text);
+  }
+
+  &.critical,
+  &.red {
+    color: var(--critical-text);
+  }
+
+  &.generic,
+  &.grey {
+    color: color.mix-token(neutral, 900);
+  }
+
+  &.generic-primary,
+  &.blue {
+    color: color.mix-token(active, 900);
+  }
+
+  &.success,
+  &.green {
+    color: var(--success-text);
+  }
+
+  &.purple {
+    color: color.mix-token(visited, 900);
+  }
+
+  &.navy {
+    color: color.mix-token(cta, 700);
+  }
+}
+
+@mixin status-background {
+  color: color.mix-token(neutral, 100);
+
+  &.info,
+  &.teal {
+    background: var(--information-background);
+  }
+
+  &.warning,
+  &.orange {
+    background: var(--warning-background);
+  }
+
+  &.critical,
+  &.red {
+    background: var(--critical-background);
+  }
+
+  &.generic,
+  &.grey {
+    background: color.mix-token(neutral, 900);
+  }
+
+  &.generic-primary,
+  &.blue {
+    background: color.mix-token(active, 900);
+  }
+
+  &.success,
+  &.green {
+    background: var(--success-background);
+  }
+
+  &.purple {
+    background: color.mix-token(visited, 900);
+  }
+
+  &.navy {
+    background: color.mix-token(cta, 700);
+  }
+}
+
+@mixin layout {
+  display: inline-block;
+
+  .text,
+  .number {
+    &.weak {
+      @include status-color;
+    }
+
+    &.strong {
+      @include status-background;
+    }
+  }
+
+  .number,
+  .dot {
+    border-radius: 50%;
+  }
+
+  .rounded {
+    border-radius: 50% !important;
```

Comment:
is there any way we could avoid using !important? We'd ideally never use this in our DS.